capsul/process/node.py

```diff
@@ -249,9 +249,16 @@ def full_name(self):
 
     def add_field(self, name, type_, default=undefined, metadata=None,
                   **kwargs):
-        # overload to add the plug
-        super().add_field(name, type_, default=default, metadata=metadata,
-                          **kwargs)
+        # delay notification until we have actually added the plug.
```

Comment:
I do not know if it is a problem here but notification is not delayed when `enable_notification=False`, events are simply not fired. We may add something in `Controller` API to store events in a queue to fire them later.